raiden/ui/web/src/app/components/deposit-dialog/deposit-dialog.component.ts

```diff
@@ -1,49 +1,50 @@
-import { Component, Inject, OnInit } from '@angular/core';
-import { FormControl } from '@angular/forms';
+import { ChangeDetectorRef, Component, Inject, OnInit, ViewChild } from '@angular/core';
 import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
-import { BigNumber } from 'bignumber.js';
+import { TokenInputComponent } from '../token-input/token-input.component';
 
 export interface DepositDialogPayload {
     decimals: number;
 }
 
+export class DepositDialogResult {
+    readonly tokenAmount: number;
+    readonly decimals: number;
+
+    constructor(
+        tokenAmount: number,
+        decimals: number
+    ) {
+        this.tokenAmount = tokenAmount;
```

Comment:
With trivial constructor, can't you use an `interface` instead of a class?

Reply:
@andrevmatos It was mostly about making the properties read only after setting them on constructor. I mean the are not supposed to change after the result goes away. If you think this is too much I can change it to an interface.

Reply:
`interface` attributes also suport the `readonly` modifier: https://www.typescriptlang.org/docs/handbook/interfaces.html [Readonly properties]